### ISSUE TITLE
Error with Predictor.deserialize

### ISSUE THREAD

**Issue by adhoc-dtn** [awslabs/gluon-ts#2167]
## Description
Hi all,
I saved trained predictor using predictor.serialize as follow.
But when I tried loading it through Predictor.deserialize,  I got the error shown below.

```python
# traning and serialize predictor
estimator = deepar.DeepAREstimator(
                                freq='M', 
                                prediction_length     = hyp["prediction_length"], 
                                context_length        = hyp["context_length"], 
                                num_layers            = hyp["num_layers"],
                                num_cells             = hyp["num_cells"],
                                use_feat_dynamic_real = True, 
                                use_feat_static_cat   = True,
                                cardinality           = [df_targets_train.index.size],
                                trainer = Trainer(batch_size=hyp["batch_size"],
                                                  clip_gradient=hyp["clip_gradient"], 
                                                  ctx='cpu',
                                                  epochs=hyp["epochs"], 
                                                  hybridize=True, 
                                                  init="xavier", 
                                                  learning_rate=hyp["learning_rate"],
                                                  learning_rate_decay_factor=hyp["learning_rate_decay_factor"],
                                                  minimum_learning_rate=hyp["minimum_learning_rate"],
                                                  num_batches_per_epoch=hyp["num_batches_per_epoch"], 
                                                  patience=hyp["patience"],
                                                  weight_decay=hyp["weight_decay"],
                                                  ), 
                                distr_output = StudentTOutput(),  
                                scaling = True,   
                                )
predictor=estimator.train(train_data)
predictor.serialize(Path("predictors/"))

# Predictor deserialize 
predictor = Predictor.deserialize(Path("predictors/"))

```

## Error message 
```
---> 2 Predictor.deserialize(Path("predictors/"))

~/anaconda3/envs/mxnet_latest_p37/lib/python3.7/site-packages/gluonts/model/predictor.py in deserialize(cls, path, **kwargs)
    120 
    121         # call deserialize() for the concrete Predictor type
--> 122         return tpe.deserialize(path, **kwargs)
    123 
    124     @classmethod

~/anaconda3/envs/mxnet_latest_p37/lib/python3.7/site-packages/gluonts/mx/model/predictor.py in deserialize(cls, path, ctx)
    394                 input_transform=transform,
    395                 prediction_net=prediction_net,
--> 396                 **parameters,
    397             )

TypeError: __init__() missing 1 required positional argument: 'freq'
```
This error was not occurred when the version of GluonTS was 0.9.5. (I don't try with version 0.9.6 and 0.10.0rc1.)
As far as I could confirm, there was not found  "freq" in parameter.json. But this item was found generated by predictor.serialize with Gluon TS version 0.9.5.
```
## GluonTS Version 0.10.0, 0.10.1, 0.10.2  parameter.json   (with error, no freq item)
{"batch_size": 32, "ctx": {"__kind__": "instance", "args": ["cpu", 0], "class": "mxnet.context.Context"}, "dtype": {"__kind__": "type", "class": "numpy.float32"}, "forecast_generator": {"__kind__": "instance", "args": [], "class": "gluonts.model.forecast_generator.SampleForecastGenerator", "kwargs": {}}, "input_names": ["feat_static_cat", "feat_static_real", "past_time_feat", "past_target", "past_observed_values", "future_time_feat", "past_is_pad"], "lead_time": 0, "prediction_length": 12}

## GluonTS 0.9.5 parameter.json  (with no error, found "freq")
{"batch_size": 32, "ctx": {"__kind__": "instance", "args": ["cpu", 0], "class": "mxnet.context.Context"}, "dtype": {"__kind__": "type", "class": "numpy.float32"}, "forecast_generator": {"__kind__": "instance", "args": [], "class": "gluonts.model.forecast_generator.SampleForecastGenerator", "kwargs": {}}, "freq": "M", "input_names": ["feat_static_cat", "feat_static_real", "past_time_feat", "past_target", "past_observed_values", "future_time_feat", "past_is_pad"], "lead_time": 0, "prediction_length": 12}


```
## Environment
- Operating system: Amazon Linux
- Python version: Python 3.7.12
- GluonTS version: 0.10.0, 0.10.1, 0.10.2
- MXNet version:1.9.1

**Comment by lostella** [comment]
@adhoc-dtn I'm having a hard time reproducing the issue with 0.10.2. Here is a minimal example.

Training/serialization:

```python
from pathlib import Path

from gluonts.dataset.common import ListDataset
from gluonts.model.deepar import DeepAREstimator
from gluonts.mx import Trainer

dataset = ListDataset(
    [
        {
            "start": "2021-01-01 00",
            "target": [1.0] * 200,
            "feat_dynamic_real": [[1.0] * 200] * 3,
        }
    ],
    freq="1H"
)

estimator = DeepAREstimator(
    freq="1H",
    prediction_length=24,
    trainer=Trainer(epochs=1),
)

predictor = estimator.train(dataset)

model_path = Path("./model")
model_path.mkdir(parents=True, exist_ok=True)
predictor.serialize(model_path)
```

Deserialization:

```python
from pathlib import Path
from gluonts.model import Predictor
model_path = Path("./model")
predictor = Predictor.deserialize(model_path)
```

No error is raised, and if I print `predictor` I get

```python
<gluonts.mx.model.predictor.RepresentableBlockPredictor at 0x13f0a7ad0>
```

**Comment by lostella** [comment]
@adhoc-dtn can you provide a minimal reproducible example that raises your error? That will help figuring out what the issue is. It needs to be self-contained code that I can copy-paste to a script or to a python session.

**Comment by jaheba** [comment]
Hello @adhoc-dtn, we've removed the `freq` argument to `ForecastGenerator` with v0.10. This might happen when you train a model with `0.10`, but try to load it using `0.9` or earlier. There should be no problem with loading it using `0.10`.

Can you briefly describe your setup?

**Comment by adhoc-dtn** [comment]
@lostella @jaheba 
Sorry I couldn't get back to you sooner. Thank you for your comments. 
I tried predictor.serialize environment and deserialize environment again with 0.10.2, no error occurred. 
When I got the error as posted, the environment where I tried "predictor.serialize" and "Predictor.deserialize" was running in separate containers. Maybe it was Gluon TS 0.10.X that ran "Predictor.serialize". But the version of GluonTS in the container that ran "Predictor.Deserialize" was 0.9.5.
I'm sorry for the trouble.
